lib/markbind/lib/parser.js

```diff
@@ -91,6 +92,12 @@ Parser.prototype._preprocess = function (node, context, config) {
   element.attribs = element.attribs || {};
   element.attribs[ATTRIB_CWF] = path.resolve(context.cwf);
   if (element.name === 'include') {
+    if (_.isEmpty(element.attribs.src)) {
+      const error = new Error(`Empty src attribute in include in: ${element.attribs[ATTRIB_CWF]}`);
+      this._onError(error);
+      element.children = cheerio.parseHTML(utils.createErrorElement(error), true);
```

Comment:
Placing the block here may not catch all empty srcs.

For example if i have this 
```html
<dynamic-panel src="' />
```

I get this error 
```shell
path.js:28
    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'path', 'string');
    ^

TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string
    at assertPath (path.js:28:11)
    at Object.resolve (path.js:221:7)
    at Parser._preprocess (D:\NUS Year 2\SEM2\markbind\markbind\lib\markbind\lib\parser.js:213:23)
    at dom.map.d (D:\NUS Year 2\SEM2\markbind\markbind\lib\markbind\lib\parser.js:359:39)
    at Array.map (<anonymous>)
    at DomHandler.htmlparser.DomHandler [as _callback] (D:\NUS Year 2\SEM2\markbind\markbind\lib\markbind\lib\parser.js:359:25)
    at DomHandler._handleCallback.DomHandler.onerror (D:\NUS Year 2\SEM2\markbind\markbind\node_modules\domhandler\index.js:52:8)
    at DomHandler.onend (D:\NUS Year 2\SEM2\markbind\markbind\node_modules\domhandler\index.js:46:7)
    at Parser.onend (D:\NUS Year 2\SEM2\markbind\markbind\node_modules\htmlparser2\lib\Parser.js:310:32)
    at Tokenizer._finish (D:\NUS Year 2\SEM2\markbind\markbind\node_modules\htmlparser2\lib\Tokenizer.js:853:12)
    at Tokenizer.end (D:\NUS Year 2\SEM2\markbind\markbind\node_modules\htmlparser2\lib\Tokenizer.js:844:25)
    at Parser.end (D:\NUS Year 2\SEM2\markbind\markbind\node_modules\htmlparser2\lib\Parser.js:338:18)
    at Parser.parseComplete (D:\NUS Year 2\SEM2\markbind\markbind\node_modules\htmlparser2\lib\Parser.js:330:7)
    at fs.readFile (D:\NUS Year 2\SEM2\markbind\markbind\lib\markbind\lib\parser.js:392:16)
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:528:3)
```

Reply:
Addressed.